vernac/prettyp.ml

```diff
@@ -425,32 +425,52 @@ let locate_any_name qid =
     try String.Map.iter iter !locatable_map; Undefined qid
     with ObjFound obj -> obj
 
+let find_location kn : Loc.t option =
+  let loc = ref None in
+  Declaremods.iter_all_segments (fun prefix c ->
```

Comment:
Declare.load_constant should put the loc in a `Loc.t Names.Cmap.t` summary ref, then you don't need to go through objects to get it

Reply:
That's a good idea, because what a mess we have here an in search. But indeed, that's still quite ad-hoc, let me think if we can come with a more principled solution for handling `load_` cases for all objects.

We really need to rework this libobject stuff, maybe we could meet about it?

Reply:
In particular we should have something that allows us to avoid `iter_all_segments` in Search too, it is possible to write a slower search implementation than the one we have, but would take some effort!

Reply:
> I don't understand what you mean.

I mean for all objects that may have a location, such as notations of inductives, we need to duplicate the code.

> Maybe it could iterate over the stuff in Global.en

I guess Search should look at objects in libobject, shouldn't it? We want to search both logical (as sent to kernel) and meta-logical stuff. WDTY?

Reply:
@SkySkimmer good point, tho still we would have to iterate the env as we do for the .vo.

Not sure how to move with the current setup to be honest, other than maybe making the kernel aware of meta-logical objects just as a container? @ppedrot what do you think?

Reply:
Please don't mess with the kernel, it's already bloated enough as it is. There is no point in registering all kinds of objects, but we should still have a generic map of "named entries" that Search can handle polymorphically and that should be extended anywhere some nametab is involved.